microsoftsccm_connector.py

```diff
@@ -80,28 +81,34 @@ def _get_fips_enabled(self):
             self.debug_print('FIPS is not enabled')
         return fips_enabled
 
-    def _execute_ps_command(self, action_result, ps_command):
-        """ This function is used to execute power shell command.
-
-        :param action_result: object of ActionResult
-        :param ps_command: power shell command
-        :return: output of executed power shell command
-        """
-
-        resp_output = None
+    def _get_protocol(self):
         server_cert_validation = 'ignore'
         transport = 'ntlm'
 
-        if self._get_fips_enabled():
+        if self._auth_type != MSSCCM_DEFAULT_AUTH_METHOD:
+            transport = self._auth_type
+        elif self._get_fips_enabled():
             transport = 'basic'
```

Comment:
I understand a logic here but in my opinion it is a bit hard to read. Maybe we can make it more verbose:

```suggestion
        if self._auth_type == MSSCCM_DEFAULT_AUTH_METHOD:
             transport = 'basic' if self._get_fips_enabled() else "ntlm"
        else:
            transport = self._auth_type
```
And then remove line with `transport = 'ntlm'` above.

Reply:
Yeah, it's much more readable that way I'll apply that